sympy/series/limits.py

```diff
@@ -247,6 +247,17 @@ def doit(self, **hints):
         if e.is_Order:
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
+        #Check to see if e is a simple power series, like (-1/2)**x
+        if e.is_Pow:
+            k = e.exp/z
+            if k.is_Number and k > 0:
+                a = abs(e.base)
+                if not a.has(z):
+                    try:
+                        if (a < 1 and z0 is S.Infinity) or (a > 1 and z0 is S.NegativeInfinity):
+                            return 0
+                    except TypeError:
+                        pass
```

Comment:
You can do this without needing to catch an exception.

Reply:
Why is this catching an exception? Is it from `a < 1` raising `TypeError` if `a` is a Symbol?

I would rather not catch `TypeError ` where possible as it can mask bugs.

Something like `(a - 1).is_extended_positive` is probably better.

Reply:
`(a - 1).is_extended_positive` doesn't work,
```python
Traceback (most recent call last):
  File "sympy_debug.py", line 4, in <module>
    f = limit(j**x, x, oo)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/limits.py", line 72, in limit
    return Limit(e, z, z0, dir).doit(deep=False)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/limits.py", line 270, in doit
    r = gruntz(e, z, z0, dir)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/gruntz.py", line 671, in gruntz
    r = limitinf(e0, z)
  File "/home/czgdp1807/sympy_project/sympy/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/gruntz.py", line 435, in limitinf
    c0, e0 = mrv_leadterm(e, x)
  File "/home/czgdp1807/sympy_project/sympy/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/gruntz.py", line 523, in mrv_leadterm
    f, logw = rewrite(exps, Omega, x, w)
  File "/home/czgdp1807/sympy_project/sympy/sympy/series/gruntz.py", line 595, in rewrite
    raise NotImplementedError('Result depends on the sign of %s' % sig)
NotImplementedError: Result depends on the sign of sign(-log(2) + I*pi)
```

Reply:
@oscarbenjamin Any views on the above comment?

Reply:
You haven't addressed the comments above:
1. This can be written without catching `TypeError`.
2. What exactly is raising the `TypeError` that is being caught here?

Reply:
I don't think `TypeError` is ever raised. However, I have observed that `NotImplementedError` is raised,
```python
NotImplementedError: Result depends on the sign of sign(log(s))
```
This happens when we pass, some input for which the following condition cannot be decided,
```python
if ((not a.has(z)) and
                        (fuzzy_and([a < 1, z0 is S.Infinity]) or
                        fuzzy_and([a > 1, z0 is S.NegativeInfinity]))):
```
The control skips the `return` under the above `if` condition and hence, the error is raised. What we can do is just return the unevaluated object if the above condition is undecidable. What do you say?

Reply:
ping @oscarbenjamin @ethankward Any comments on the above?